retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts

```diff
@@ -1,40 +1,73 @@
-import { CategoryRepository } from '../../domain/category/category.repository';
 import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { CategoryRepository } from '../../domain/category/category.repository';
 import { Category } from 'src/app/domain/category/models/category';
-import { MatDialog } from '@angular/material/dialog';
-import { PopUpComponent } from './pop-up/pop-up.component';
 @Component({
   selector: 'app-category-page',
   templateUrl: './category-page.component.html',
+  styles: [
+    '.category { min-height: auto; } table { min-width: 1100px; min-height: auto; }mat-icon{ font-size: 29px;} .btn {background-color: #002d40; color: white;}',
+  ],
 })
 export class CategoryPageComponent implements OnInit {
+  categoryform!: FormGroup;
   allCategory: Category[] = [];
   isAppear!: boolean;
-  displayedColumns: string[] = ['id', 'name', 'status'];
-  dataimage: any;
+  displayedColumns: string[] = ['id', 'name', 'status', 'update', 'delete'];
   submit: boolean = false;
   submitted: boolean = false;
+  data!: Category;
```

Comment:
are you used this property?
line 18 and 19?

Reply:
Submitted used in onsubmit function, but data!:Category deleted